### ISSUE TITLE
[MINOR] Rules can be created without a deployment channel being configured

### ISSUE THREAD

**Issue by pbartusch** [querqy/smui#95]
Steps to reproduce:

* Start SMUI on a fresh database
* Create Rules

Actual:

* Rules can be created (but the assignment to a deployment channel is unclear, also they disappear once a deployment channel is configured properly - which is unexpected to the user).

Expected:

* Deliver a warning to the user.

**Comment by epugh** [comment]
Thanks for opening this ticket...   I was a bit confused if there was an idea that if a rule doesn't have a Deployment Channel (Rules Collection?) then maybe it applied to all of them.   THanks for confirming that rules must be tied to a Deployment Channel.   

As an aside, I think we need to get on the same page for this name!

**Comment by pbartusch** [comment]
Yeah. I think, rules will created with an empty string reference then.

Sorry for the confusion, I think renaming the SolrIndex concept and discussing the alternatives (Deployment Channel vs. Rules Collection) is an important topic (on its own). It's inconsistent through code, documentation and conversions right now :-(

**Comment by epugh** [comment]
Once we get a clear name, I'm happy to go do the refactoring...   (Refactoring is my favorite Friday afternoon activity!)
